msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * (C) Copyright 2015-2018 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+import de.unijena.bioinf.ChemistryBase.chem.ChemicalAlphabet;
+import de.unijena.bioinf.ChemistryBase.chem.Element;
+import de.unijena.bioinf.ChemistryBase.chem.FormulaConstraints;
+import de.unijena.bioinf.ChemistryBase.chem.PeriodicTable;
+import de.unijena.bioinf.ChemistryBase.chem.PrecursorIonType;
+import de.unijena.bioinf.ChemistryBase.ms.Deviation;
+import de.unijena.bioinf.ChemistryBase.ms.MutableMs2Experiment;
+import de.unijena.bioinf.ChemistryBase.ms.MutableMs2Spectrum;
+import de.unijena.bioinf.ChemistryBase.ms.Peak;
+import de.unijena.bioinf.ChemistryBase.ms.Spectrum;
+import de.unijena.bioinf.ChemistryBase.ms.utils.SimpleSpectrum;
+import de.unijena.bioinf.sirius.IdentificationResult;
+import de.unijena.bioinf.sirius.IsotopePatternHandling;
+import de.unijena.bioinf.sirius.Sirius;
+import io.github.msdk.MSDKException;
+import io.github.msdk.MSDKMethod;
+import io.github.msdk.MSDKRuntimeException;
+import io.github.msdk.datamodel.IonAnnotation;
+import io.github.msdk.datamodel.IonType;
+import io.github.msdk.datamodel.MsSpectrum;
+import io.github.msdk.datamodel.MsSpectrumType;
+import io.github.msdk.datamodel.SimpleIonAnnotation;
+import io.github.msdk.datamodel.SimpleMsSpectrum;
+import io.github.msdk.spectra.centroidprofiledetection.SpectrumTypeDetectionAlgorithm;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.openscience.cdk.formula.MolecularFormulaRange;
+import org.openscience.cdk.interfaces.IIsotope;
+import org.openscience.cdk.interfaces.IMolecularFormula;
+import org.openscience.cdk.silent.SilentChemObjectBuilder;
+import org.openscience.cdk.tools.manipulator.MolecularFormulaManipulator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p> SiriusIdentificationMethod class. </p>
+ *
+ * This class wraps the Sirius module and transforms its results into MSDK data structures
+ * Transformation of IdentificationResult (Sirius) into IonAnnatation (MSDK)
+ * Containts sub-class for generating FormulaConstraints for Sirius from MolecularFormulaRange object
+ */
+public class SiriusIdentificationMethod implements MSDKMethod<List<IonAnnotation>> {
+
+  /**
+   * Dynamic loading of native libraries
+   */
+  static {
+    try {
```

Comment:
It seems a bit strange to use System.load() and then catch the UnsatisfiedLinkException. Perhaps this whole logic should be moved to the NativeLibraryLoader class.

Reply:
I decided to remove the old code and it is now fully controlled by native library loader.